### ISSUE TITLE
{} and + shift up/down between regular and italics

### ISSUE THREAD

**Issue by ciscorucinski** [chrissimpkins/Hack#141]
![output_jxcuoz](https://cloud.githubusercontent.com/assets/8707125/10824496/fee273e2-7ea3-11e5-93d3-5ce7bf4f4c80.gif)

Check out line 437 in the picture

I am switching between regular and italic comments. On line 437, you can see `{ }` and `+` shift up and down. That should not happen with italics. I don't see any other issues with shifts up and down.

This also seems to happen with all font sizes

**Comment by chrissimpkins** [comment]
Will you also try this in a non Jetbrains editor?  In some of their editors, they implement their own obliques from the regular set rather than using the oblique/italic set.  I think that we fixed this issue with the name change and it does not appear to be a factor here because that looks like the proper Hack oblique slant, but let's be sure.

**Comment by chrissimpkins** [comment]
The jumping `+` is a hinting problem and we are addressing that in #138.

**Comment by chrissimpkins** [comment]
The `{` + `}` are aligned in different vertical positions in the design.

<img width="788" alt="braces" src="https://cloud.githubusercontent.com/assets/4249591/10825280/bcfce5ca-7e3a-11e5-954b-a4af0f7c8f76.png">

This isn't something that I've noticed previously.  Thanks for submitting this issue report.  The oblique variants do appear a bit low in the box vs. the adjacent glyphs (see small text in the window behind the focus window in the image above).  The question is whether that is desirable or not in the context of source code which is a bit more complex question than whether they have the same alignment between sets.  We are in the process of performing a detailed analysis of source code tokens and collocalized glyphs/tokens to push out additional optimizations in the fonts.  With the prevalence of braces in commonly used source languages, they are definitely high on our list.  Will be in touch in this thread as we examine the issue and propose changes.  

Out of interest, what source type do you commonly work in and are the braces typically immediately adjacent to alphabetic glyphs or to spaces?

**Comment by ciscorucinski** [comment]
what do you mean source type? Languages?

**Comment by chrissimpkins** [comment]
Yes, sorry - what programming language(s) are you using.  We are looking at several of the most commonly used as our initial optimization targets.  The punctuation glyphs are used in a different fashion across languages (e.g. {} define code blocks in JS, C, used for Python dictionaries) and this is all being taken into consideration.

**Comment by ciscorucinski** [comment]
I normally use Java (Android developement), but also use C#. 

Usually the brackets are kind of separated from text, but I usually have single-line methods that are short enough.

Personally, with customizations allowed in some IDEs, I feel like the they should share the same baseline. You never know what a person would like, and if these could cause them to look not so great.

**Comment by ciscorucinski** [comment]
I have my current scope braces use bold style, and all other braces use regular style. 

I noticed that the `{ }` seem to have the same baseline, but the bold braces are smaller in height. Same with bold-italics.

I will do more testing to verify this.

**Comment by ciscorucinski** [comment]
Regular vs. Bold style. I see there are more issues here.
![output_m0bv2l](https://cloud.githubusercontent.com/assets/8707125/10827188/e5bcc77a-7eb0-11e5-84b8-c315050a1b17.gif)

**Comment by ciscorucinski** [comment]
different sizes do cause extra issues.

For example, in size 12, the tilda (~) does not move like it does in size 10. However, in size 12, the underbar (_) moves down compared to its position in size 10.

There are more issues, just giving examples.

**Comment by chrissimpkins** [comment]
I am going to put you in touch with @burodepeper as well.  I think that the two of you will find common ground in these conversations and he is leading the usability work that will provide the basis for the changes that I mentioned.  Let's keep this conversation going.

**Comment by ciscorucinski** [comment]
@burodepeper if you want me to do other animated gifs with other styles, then let me know

While I didn't do extensive tests on either from `regular` to `bold or italics`, it seems like the italics issues are font-size independent. However, the bold issues are font-size dependent. 

A few more tests will have to be done to see if the italic issues are actually font-size dependent

**Comment by burodepeper** [comment]
@ciscorucinski Hey! First of all, thanks for your effort so far.

I think I have enough information for now to go on. My main focus is currently on `regular`, and any changes there will flow back/forward into italic/bold. Strange enough, I've never noticed the misalignment of the curly braces before, even though I stare at them most of the day.

I'll let you know when we have an update for you to find new bugs in! ; )

Don't bother extensively testing various font-sizes etc for now (unless you really want to), but if you happen to run into any other issues that occur naturally in the wild, please let us know.

**Comment by chrissimpkins** [comment]
@burodepeper @ciscorucinski :+1:

**Comment by ciscorucinski** [comment]
Glad to help. I will wait for that update.

**Comment by chrissimpkins** [comment]
@burodepeper IR associated with braces work
